R/nest_by.R

```diff
@@ -0,0 +1,57 @@
+#' Nest by one or more variables
+#'
+#' @description
+#' \Sexpr[results=rd, stage=render]{lifecycle::badge("experimental")}
+#'
+#' `nest_by()` is closely related to [group_by()]. However, instead of storing
+#' the group structure in the metadata, it makes it explicit in the data,
+#' giving each group key a single row with a list-column data frames that
+#' contains all the other data.
```

Comment:
```suggestion
#' giving each group key a single row with a list-column of data frames that
#' contain all the other data.
```

Will need to redocument (this might be a nice use of the github actions document action)